### ISSUE TITLE
Completion Metadata

### ISSUE THREAD

**Issue by DerWeh** [wilywampa/vim-ipython#1]
I would like to use the IPython auto-complete via `vim-ipython`. According to your documentation I added
```
def setup_kernel():
    request = r"""
import inspect
import IPython
def completion_metadata(ip):
    metadata = [dict(word=m) for m in ip.Completer.matches]
    for m in metadata:
        try:
            obj = eval(m['word'], ip.user_ns)
        except Exception:
            continue
        doc = inspect.getdoc(obj)
        if callable(obj):
            try:
                m['menu'] = str(IPython.utils.signatures.signature(obj)) + '\t'
            except Exception:
                m['menu'] = ''
            if doc:
                m['menu'] += doc.split('\n')[0]
                m['info'] = doc
        else:
            m['menu'] = str(obj)
            if doc:
                m['info'] = doc
    return metadata
"""
    send(request, silent=True)
```
 
This works more or less. But I would like to set a different `word` and `abbr` to get the function signature nicely without appending it (similar as in [http://www.vim.org/scripts/script.php?script_id=1542](http://www.vim.org/scripts/script.php?script_id=1542)). They just use separate dictionary entries for this. Adding this in the python snippet is no problem, but I have no idea how to handle this in vimscript. Is this somehow possible?

**Comment by wilywampa** [comment]
If I'm understanding the problem, it sounds like the right approach might be to replace the code [here and pasted below ](https://github.com/wilywampa/vim-ipython/blob/master/ftplugin/python/ipy.vim#L281) with something that makes a Vim dictionary with every key in the Python dictionary instead of just checking for an `info` key.

``` python
        if 'info' in m:
            result.menu, result.info = m['menu'], m['info']
            vim.command('call add(res, {"word": IPythonPyeval("r.word"),    '
                                       '"menu": IPythonPyeval("r.menu"), '
                                       '"info": IPythonPyeval("r.info")})')
        else:
            result.menu = m.get('menu', '')
            vim.command('call add(res, {"word": IPythonPyeval("r.word"),    '
                                       '"menu": IPythonPyeval("r.menu")})')
```

That way if you add an `abbr` (or `kind`, etc.) key in `completion_metadata()` Vim would see it automatically. Would that help?

**Comment by wilywampa** [comment]
Can you try [this branch](https://github.com/wilywampa/vim-ipython/tree/metadata_dict) and check if using `abbr` works as you'd like?

**Comment by DerWeh** [comment]
Apparently my answer wasn't submitted.
It works like a charm, thank you for your splendid and incredible fast fix. Now I can try customizing it again. The plug-in provides the superior alternative to Jedi in my opinion as the auto completion is fast enough. One more question if you allow:
You mentioned further keys like `kind`. I looked for them but couldn't find anything. Do you by chances have a reference or documentation which options do exist?

**Comment by wilywampa** [comment]
Sure, `:help complete-items` will take you right to it.

**Comment by wilywampa** [comment]
Closed via 9dd97ecb2acec9557c7350eacdda9d02f5f40ba0.
